File: lib/csharp/microsoft.bot.builder.skills/Microsoft.Bot.Builder.Skills/SkillTransport.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Threading;
using System.Threading.Tasks;
using Microsoft.Bot.Builder.Skills.Auth;
using Microsoft.Bot.Builder.Skills.Models.Manifest;
using Microsoft.Bot.Schema;

namespace Microsoft.Bot.Builder.Skills
{
    public abstract class SkillTransport
    {
        public abstract Task<Activity> ForwardToSkillAsync(ITurnContext turnContext, SkillManifest skillManifest, IServiceClientCredentials serviceClientCredentials, Activity activity, ISkillResponseHandler skillResponseHandler, CancellationToken cancellationToken = default);

        public abstract Task CancelRemoteDialogsAsync(ITurnContext turnContext, SkillManifest skillManifest, IServiceClientCredentials serviceClientCredentials, CancellationToken cancellationToken = default);

        public abstract void Disconnect();
    }
}

File: sdk/typescript/libraries/botbuilder-skills/src/skillTransport.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import { TurnContext } from 'botbuilder';
import { Activity } from 'botframework-schema';

export type TokenRequestHandler = (activity: Activity) => Promise<void>;
export type FallbackHandler = (activity: Activity) => Promise<void>;

export interface ISkillTransport {
    // PENDING: both method should be uncommented when the WS library is merged
    // forwardToSkill(
    //     skillManifest: ISkillManifest,
    //     serviceClientCredentials: IServiceClientCredentials,
    //     turnContext: TurnContext,
    //     activity: Partial<Activity>,
    //     tokenRequestHandler?: TokenRequestHandler,
    //     fallbackHandler?: FallbackHandler
    // ): Promise<Partial<Activity>>;
    // cancelRemoteDialogs(skillManifest: ISkillManifest, appCredentials: IServiceClientCredentials, turnContext: TurnContext): Promise<void>;
    forwardToSkill(turnContext: TurnContext, activity: Partial<Activity>, tokenRequestHandler?: TokenRequestHandler): Promise<boolean>;
    cancelRemoteDialogs(turnContext: TurnContext): Promise<void>;
    disconnect(): void;
}
